src/opencl_rawsha256_fmt_plug.c

```diff
@@ -304,18 +309,21 @@ static void reset(struct db_main *db)
 		                       warn, 1, self, create_clobj, release_clobj,
 		                       2 * BUFFER_SIZE, gws_limit);
 
+		if ((options.flags & FLG_MASK_CHK) && (options.flags & FLG_TEST_CHK)) {
+			autotune_limit = opencl_get_tune_time_limit(gpu_id, autotune_limit);
+		}
 		//Auto tune execution from shared/included code.
-		autotune_run(self, 1, gws_limit, 500ULL);
+		autotune_run(self, 1, gws_limit, autotune_limit);
 
 		load_hash(NULL);
 
-		if (options.flags & FLG_MASK_CHK) {
+		if ((options.flags & FLG_MASK_CHK) && !(options.flags & FLG_TEST_CHK)) {
 			fprintf(stdout,
 			        "Using Mask Mode with internal candidate generation%s",
 			        flag ? "" : "\n");
 
 			if (flag) {
-				self->params.max_keys_per_crypt /= 256;
+				self->params.max_keys_per_crypt /= 32;
 
```

Comment:
It works correctly with eg. nt-opencl (which uses our shared auto-tune). Check for clues in reset().